### DISCUSSION TITLE
Special header folding when using PHP's mail function still required?

### DISCUSSION THREAD

**Discussion by daeppen** [PHPMailer/PHPMailer#3022] [General]
To my understanding, special handling regarding header folding when using `PHPMailer::isMail()` (`mail()`) has been introduced in https://github.com/PHPMailer/PHPMailer/commit/b25f93eb2d7650aa5cc1b144d61e8526c0c88257 and https://github.com/PHPMailer/PHPMailer/commit/21b35dc49b9377befdb6883014d04233ceedf78e along with the constant `MAIL_MAX_LINE_LENGTH` due to #818.
I'm curious if this is still an issue (by the `mail()` function) in recent PHP versions? Currently I'm unable to reproduce the issue using PHP 8.2.15. Meaning when I set `MAIL_MAX_LINE_LENGTH` to `998` the headers are not broken.
Is anyone able to reproduce the issue which would still legitimate the existence of `MAIL_MAX_LINE_LENGTH`?

Background: I'm trying to implement `List-Unsubscribe`. My tests show that certain mail clients (tested with Thunderbird, FairEmail & eM Client) seem not yet to support folded `List-Unsubscribe` headers. Dropping `MAIL_MAX_LINE_LENGTH` or increasing it to `998` prevents folding on `List-Unsubscribe` in my use cases.

**Comment by Synchro** [comment]
Indeed I've not been able to repro this issue myself, but "works for me" isn't a great defence! Overall I always recommend using SMTP to localhost instead of using mail(). It's safer, faster, and provides better debug feedback.

With the recent revelation in #3012 that PHP may have changed its policy on line breaks in the mail function, I see even fewer reasons to support it as it makes legacy support increasingly difficult.

**Reply by daeppen** [reply]
Thanks for your quick reply. I understand. So I'll switch to SMTP for good.

**Comment by campbell-m** [comment]
If you're thinking of dropping support for mail() then I'm sure you'll appreciate that there are many existing sites using mail() - however inadvisedly - for whom that would be a shock if it were suddenly dropped without some kind of graceful transition path to SMTP.

**Reply by Synchro** [reply]
Oh, I'm not thinking of dropping it, but I will keep recommending not using it!

**Reply by campbell-m** [reply]
OK!
